dev-docs/services/ci/README.md

```diff
@@ -0,0 +1,244 @@
+- The `WatchedBranch`'s `_update` method in [`github.py`](../../../ci/ci/github.py) iterates again to merge all mergeable PRs, in priority order
+
+## Deploying services to the live infrastructure
+
+When a PR is merged into the `main` branch, a webhook will trigger. The CI service will set its `github_changed` flag.
+
+During its update loop, the CI service will determine that the SHA of the `WatchedBranch` has changed and trigger
+a deployment.
+
+- Create a new batch job to:
+  - Build various components and service images
+  - Run various pre-deployment tests
+  - Deploy to the `default` (ie prod) namespace
+  - Run various post-deployment tests
+  - A handful of final actions
+    - eg rolling out artifact registry cleanup policies, amongst many other things
+
+Note: It's not actually quite as waterfall-y as this. In fact the jobs are all running in a hail
+batch, and each package being built and service being deployed has its own path through the DAG. So it's quite possible
+that services are deploy/test-ing in parallel, and that the deploy for one service might happen before the test for
+another has completed.
+
+This should all be fine, because everything was previously tested as part of the PR approval process.
+
+Examples of CI deploy runs can be seen by searching through the production batch log, as long as you have developer
+permissions. For example: `/batches?q=user+%3D+ci%0D%0Adeploy+%3D+1`
+
+#### CI Deploy Timeline
+
+The image below shows the CI deployment timeline:
+
+```mermaid
+sequenceDiagram
+    participant Github
+    box Kubernetes (PROD: 'default' namespace)
+    participant CI as PROD CI Service (default namespace)
+    participant PB as PROD Batch (default namespace)
+    end
+    box GCE (PROD VMs)
+    participant PVM as  PROD VMs
+    end
+
+    Github->>CI: Notify 'main' branch updated
+    CI->>PB: Submit CI deploy batch
+    activate PB
+    PB->>PVM: Submit build jobs
+    activate PVM
+    PVM-->>PB: Done
+    deactivate PVM
+    PB->>PVM: Submit deploy jobs
+    activate PVM
+    PVM->>PB: Redeploy batch service
+    PVM->>CI: Redeploy CI service
+    PVM-->>PB: Done
+    deactivate PVM
+    PB->>PVM: Submit test jobs
+    activate PVM
+    PVM->>PB: Submit test batches
+    PB->>PVM: Submit test batch jobs
+    activate PVM
+    PVM-->>PB: Validate results
+    deactivate PVM
+    PVM-->>PB: Done
+    deactivate PVM
+    PB-->>CI: Detects completion
+```
+
+## Issues
+
+### Hanging PR state
+
+It is sometimes possible for a PR status to be stuck in pending even though the test batch has completed
+This can be fixed by pushing a new commit to the branch, but it's strange that regular polling does not catch
+these instances.
+ 
+## Open Questions
+
+1. Sometimes in logs we see logs like `"update github br-hail-ci-test-ci-test-<RANDOM>-main"` for various random branch names. 
+What does this mean?
```

Comment:
Could these branches be the results of squashing and rebasing a PR before running CI?